e2e/LongHaul/device/Program.cs

```diff
@@ -104,13 +105,36 @@ private static Logger InitializeLogging(string deviceConnectionString, string ai
 
         private static IotHub InitializeHub(Logger logger)
         {
-            var iotHub = new IotHub(logger, _settings.DeviceConnectionString, _settings.Transport);
+            var iotHub = new IotHub(logger, _settings.DeviceConnectionString, GetTransportSettings(_settings.TransportProtocolType));
             foreach (var prop in _commonProperties)
             {
                 iotHub.IotProperties.Add(prop.Key, prop.Value);
             }
 
             return iotHub;
         }
+
+        private static IotHubClientTransportSettings GetTransportSettings(string transportProtocolType)
```

Comment:
Previously the setting file takes the property "Transport" in type of `TransportType`. Now that we don't have this type in V2 any more, change the property type into `string` and map it to the specific `IotHubClientTransportSettings` object here.

Reply:
Or make an enum like we do for many of our samples. It'll parse better that way.

Reply:
https://github.com/Azure/azure-iot-sdk-csharp/blob/previews/v2/iothub/device/samples/how%20to%20guides/DeviceReconnectionSample/Parameters.cs#L9

And change to 2 params, one for transport and one for transport protocol.